barretenberg/cpp/pil/vm2/execution.pil

```diff
@@ -229,28 +251,44 @@ pol commit rop[7];
 // The "output we want is" sel_addressing_error from addressing.pil.
 
 /**************************************************************************************************
- *  Temporality group N: Registers + Dispatch opcode
+ *  Temporality group 3: Register read.
  **************************************************************************************************/
 
-// TODO: gate this group.
+pol commit sel_should_read_registers;
+sel_should_read_registers = sel_should_resolve_address * (1 - sel_addressing_error);
 
 // Registers
 pol commit register[7];
 // Memory Acccesses
-pol commit mem_op[7];
-// Memory Tag
-pol commit mem_tag[7];
+pol commit sel_mem_op_reg[7];
 // Read / Write selectors
-pol commit rw[7];
+pol commit rw_reg[7];
+// Memory Tag
+pol commit mem_tag_reg[7];
+// Whether we should perform a tag check on the register
+pol commit sel_tag_check_reg[7];
+// Expected tag
+pol commit expected_tag_reg[7];
+
+// NOTE: Constraints on the registers are in execution_registers.pil.
+// The "output we want is" sel_register_read_error from execution_registers.pil.
+
+/**************************************************************************************************
+ *  Temporality group 4: Gas (will be moved here).
+ **************************************************************************************************/
+
+// TODO(fcarreiro): Unify gas and move here.
 
 /**************************************************************************************************
- *  Temporality group N: Dispatch opcode
+ *  Temporality group 5: Opcode execution
  **************************************************************************************************/
 
-// TODO: gate this group with some error flag from the previous group.
+pol commit should_execute_opcode;
+should_execute_opcode = sel_should_read_registers * (1 - sel_register_read_error);
 
-pol commit opcode_error;
-opcode_error * (1 - opcode_error) = 0;
+pol commit sel_opcode_error;
+#[OPCODE_ERROR_BOOLEAN]
```

Comment:
Reintroduction of `#[OPCODE_ERROR_BOOLEAN]` is a merge error. I'll remove it if I have to make other changes.